torch/csrc/jit/script/compiler.cpp

```diff
@@ -1706,36 +1738,22 @@ struct to_ir {
             {},
             stmt.range());
       }
-    } else {
-      // Otherwise, it should be a list.  Lower this expression into:
+    } else if (
+        const ListTypePtr listType = sliceable->type()->cast<ListType>()) {
+      // If it's a list.  Lower this expression into:
       //     list.set_item(get_item(idx).add_(value))
       // similar to how Python handles things.
-      const auto listType = sliceable->type()->cast<ListType>();
-      AT_ASSERT(listType != nullptr);
-
       auto elementType = listType->getElementType();
-
-      // Get the idx to augment
-      const auto subscriptExprs = lhs.subscript_exprs();
-      if (subscriptExprs.size() != 1) {
-        throw ErrorReport(subscriptExprs)
-            << "Sliced expression not yet supported for"
-            << " subscripted list augmented assignment. "
-            << "File a bug if you want this";
-      }
-      const auto idxValue = emitExpr(subscriptExprs[0]);
-
-      const auto listArg = NamedValue(lhs.value().range(), "list", sliceable);
-      const auto idxArg = NamedValue(subscriptExprs.range(), "idx", idxValue);
-      const auto valueArg =
-          NamedValue(stmt.rhs().range(), "value", emitExpr(stmt.rhs()));
-
-      const auto getItem =
-          graph->insert(aten::__getitem__, {listArg, idxArg}, {}, stmt.range());
-      const auto augmentedItem = graph->insert(
-          getAugOp(stmt, elementType), {getItem, valueArg}, {}, stmt.range());
-      graph->insert(
-          aten::_set_item, {listArg, idxArg, augmentedItem}, {}, stmt.range());
+      emitAugAssignToContainer("list", elementType, stmt, lhs, sliceable);
```

Comment:
Similarly, you don't need this branching here—it can just be `emitAugAssignGeneric` and take a type. Internally that function can do the check that the type is supported